### PR TITLE
WT-7789 Change tiered python tests to fail without extension library

### PR DESCRIPTION
**Pull request by keitharnoldsmith**
Quick change so the tiered tests will fail rather than skip if the `local_store` library isn't found.

I removed `libwiredtiger_local_store.so` in my tree and confirmed that the tests complain about it.

### CODE REVIEW

**Review by ddanderson — APPROVED** [review]
LGTM

**Comment by keitharnoldsmith** [comment]
The Windows PR testing is failing because the tiered tests can't find the tiered library. Apparently we either don't build the extensions on Windows or put them someplace different.

**Review by sueloverso — APPROVED** [review]
LGTM although I don't know that part of the python code.

**Comment by keitharnoldsmith** [comment]
@ddanderson & @sueloverso, I'm pinging you for re-review on this one.

I've updated the tiered python tests to conditionally decide whether to skip or fail if they can't find the `local_store` extension library.  On most systems we'll fail if it isn't there. This way we won't have silent errors if the build (or something else) goes wrong. But since we don't support dynamic loading of extension libraries on Windows, we will silently skip these tests there.

I confirmed that the tests pass on Linux are skipped on Windows and that they fail on Linux when I manually remove `libwiredtiger_local_store`.

File: test/suite/test_tiered02.py [hunk 1]
```diff
@@ -48,9 +48,11 @@ def conn_config(self):
           'bucket_prefix=%s,' % self.bucket_prefix + \
           'name=%s),tiered_manager=(wait=0)' % self.extension_name
 
-    # Load the local store extension, but skip the test if it is missing.
+    # Load the local store extension. 
     def conn_extensions(self, extlist):
-        extlist.skip_if_missing = True
+        # Windows doesn't support dynamically loading extension libraries.
```
Comment (sueloverso) [inline test/suite/test_tiered02.py:53]: This sentence is correct as-is. But when I first read it I expected slightly different wording. Ignore this comment if you want. But either of these are what I expected to read: (if you change it, please change it in all the files)

Windows doesn't support dynamically loaded extension libraries.
Windows doesn't support loading extension libraries dynamically.

**Review by sueloverso — APPROVED** [review]
LGTM with one potential wording suggestion.

**Review by ddanderson — APPROVED** [review]
LGTM.

We might also consider having a base class for the tiered tests that handles this boiler plate stuff.

File: test/suite/test_tiered02.py (hunk 1 above)
Comment (keitharnoldsmith) [reply test/suite/test_tiered02.py]: Done.  (Using the first option)

**Comment by keitharnoldsmith** [comment]
@ddanderson, I'll create a separate ticket to refactor the common elements of tiered test cases into a common base class.

**Pull request merged**